writer.go

```diff
@@ -0,0 +1,145 @@
+// Package sstable implements the reading and write of SSTable files.
+//
+// SSTable file format is defined here:
+// https://www.igvita.com/2012/02/06/sstable-and-log-structured-storage-leveldb/
+// In particular, a basic SSTable file contains a sequence of records,
+// where each record consists of:
+//
+//  1. key-size in 4 bytes of little-endian uint32
+//  2. key
+//  3. value-size in 4 bytes of little-endian uint32
+//  4. value
+//
+// An SSTable file can optionally has the index data structured
+// appended to records:
```

Comment:
In the referenced article:
> Optionally, if the file is very large, we can also prepend, or create a standalone key:offset index for fast access.

I think prepend would be better than append, since by doing so, reading the index would not need to seek to the end of the file. I am not totally sure, but maybe seek to the end of the file in the distributed filesystem is a heavy operation (I created a question on stackoverflow: http://stackoverflow.com/questions/43686824/network-usage-of-file-open-and-seek-in-distributed-filesystem-like-ceph).

I understand prepend to the file requires knowing the size of index section, and it's hard to know it beforehand. We could consider either only allow knowing it beforehand (in Bigtable, SSTable is created from Memtable, it knows the index section size), or write it to a different file on disk.